gdpr_assist/admin/base.py

```diff
@@ -81,23 +110,18 @@ def anonymise_view(self, request):
         )
 
         if request.POST:
-            objects.anonymise(user=request.user)
+            self.perform_anonymisation(request, objects)
             count = objects.count()
-            messages.success(
-                request,
-                "{} {} anonymised".format(count, verbose_name),
-            )
+            self.success_message(request, verbose_name, count)
             return HttpResponseRedirect(changelist_url)
 
-        object_classes = set([o.__class__ for o in objects])
         tree_html = ""
-        for object_class in object_classes:
```

Comment:
how are we removing this loop?

Reply:
Won't all the top objects be of the same model class since it's a `ModelAdmin` page?

Reply:
generally yes, but not always. exception: that `PersonalData` page you can select multiple types of objects on (also maybe the `RetentionPolicy` one, but i forgot the details of how i did that there

Reply:
The personal data admin page doesn't use that, it doesn't have a confirmation page. For any other model inheriting this `ModelAdmin` class, it will still be one type of model for the top objects you select.